staging/src/k8s.io/apiserver/pkg/storage/etcd3/preflight/checks.go

```diff
@@ -59,9 +60,12 @@ func parseServerURI(serverURI string) (*url.URL, error) {
 // CheckEtcdServers will attempt to reach all etcd servers once. If any
 // can be reached, return true.
 func (con EtcdConnection) CheckEtcdServers() (done bool, err error) {
-	// Attempt to reach every Etcd server in order
-	for _, serverURI := range con.ServerList {
-		host, err := parseServerURI(serverURI)
+	rand.Seed(time.Now().UTC().UnixNano())
+	// Attempt to reach every Etcd server randomly.
+	serverNumber := len(con.ServerList)
+	serverPerms := rand.Perm(serverNumber)
```

Comment:
what is the cost of this permutation per health check ?

Reply:
This can be ignored.
```
// Perm returns, as a slice of n ints, a pseudo-random permutation of the integers [0,n).
func (r *Rand) Perm(n int) []int {
	m := make([]int, n)
	// In the following loop, the iteration when i=0 always swaps m[0] with m[0].
	// A change to remove this useless iteration is to assign 1 to i in the init
	// statement. But Perm also effects r. Making this change will affect
	// the final state of r. So this change can't be made for compatibility
	// reasons for Go 1.
	for i := 0; i < n; i++ {
		j := r.Intn(i + 1)
		m[i] = m[j]
		m[j] = i
	}
	return m
}
```